sale_coupon_apply_commercial_partner/models/coupon_coupon.py

```diff
@@ -0,0 +1,17 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, models
+
+
+class SaleCoupon(models.Model):
+    _inherit = "sale.coupon"
+
+    def _check_coupon_code(self, order):
+        message = super()._check_coupon_code(order)
+        if message.get("error", "") == _("Invalid partner."):
+            if (
+                self.partner_id.commercial_partner_id
+                == order.partner_id.commercial_partner_id
+            ):
+                message.pop("error")
```

Comment:
```suggestion
                message.pop("error", None)
```

Reply:
Not necessary, as the previous `if` makes sure we've got the key.